marklogic-data-hub-central/ui/src/components/modeling/entity-type-table/entity-type-table.tsx

```diff
@@ -37,7 +37,7 @@ const EntityTypeTable: React.FC<Props> = (props) => {
 
   const [showConfirmModal, toggleConfirmModal] = useState(false);
   const [confirmBoldTextArray, setConfirmBoldTextArray] = useState<string[]>([]);
-  const [stepValuesArray, setStepValuesArray] = useState<string[]>([]);
+  const [arrayValues, setArrayValues] = useState<string[]>([]);
   const [confirmType, setConfirmType] = useState<ConfirmationType>(ConfirmationType.DeleteEntity);
   const [isAnyEntityModified, toggleAnyEntityModified] = useState(modelingOptions.isModified);
```

Comment:
I don't think this needs to be it's own state variable. You should use 'modelingOptions.isModified' directly and use the context functions to do mutations

Reply:
This isn't introducing a new state variable. It's just renaming the existing one from stepValuesArray to arrayValues since we are using that to pass both steps and model names.

Reply:
Sorry, I was actually referring to "isAnyEntityModified". You can use "modelingOptions.isModified" directly instead. See "pages/Modeling.tsx" if you need an example

Reply:
I had to add `isAnyEntityModified` as a state variable so that react would update the save and revert icons based on the operations performed on both the parent and the current component.
So when we revert all / save all from the parent component then the state change makes React to rerender the save/revert icons on the entity-type-table page.
```
className={(!props.canWriteEntityModel && props.canReadEntityModel) || !isAnyEntityModified || !isEntityModified(text) ? styles.iconSaveReadOnly : styles.iconSave}
```

Reply:
Ok got it. That makes sense.